server/src/main/java/org/elasticsearch/search/aggregations/metrics/TopHitsAggregator.java

```diff
@@ -196,7 +197,12 @@ public InternalAggregation buildAggregation(long owningBucketOrdinal) throws IOE
         for (int i = 0; i < topDocs.scoreDocs.length; i++) {
             docIdsToLoad[i] = topDocs.scoreDocs[i].doc;
         }
-        FetchSearchResult fetchResult = runFetchPhase(subSearchContext, docIdsToLoad);
+        FetchSearchResult fetchResult = runFetchPhase(
```

Comment:
Does it even make any sense to do the accounting here? We literally only account while accumulating the response, then we create the top hits result which moves the hits to `unpooled` and release the breaker count again without saving a single byte of heap? Seems like pointless overhead?

Reply:
>We literally only account while accumulating the response

Reply:
Yea but the releasing of the refcounted happens right after accumulating the heap usage, so we literally do the breaker release at peak heap usage, making the whole exercise somewhat redundant?

Reply:
>Sure we may break while going up

I believe this is a valid use case though (parallel collections leading to OOM is something we saw)